network/src/main/scala/com/linkedin/norbert/network/server/MessageHandlerRegistryComponent.scala

```diff
@@ -21,47 +21,73 @@ trait MessageHandlerRegistryComponent {
   val messageHandlerRegistry: MessageHandlerRegistry
 }
 
-private case class MessageHandlerEntry[RequestMsg, ResponseMsg]
+private case class SyncHandlerEntry[RequestMsg, ResponseMsg]
```

Comment:
Maybe we can introduce generic HandlerEntry with case classes that can keep code more cleaner. Inside of the MessageHanlderRegistryComponent, try to handle selecting sync and async cases as much as possible to keep client side. Thus, we can also remove to check whether a given message is sync or async types.